utils.pwsh/Setup-Target.ps1

```diff
@@ -0,0 +1,95 @@
+function Setup-Target {
+    if ( ! ( Test-Path function:Log-Output ) ) {
+        . $PSScriptRoot/Logger.ps1
+    }
+
+    $Target64Bit = ( $script:Target -eq 'x64' )
+
+    $script:ConfigData = @{
+        Arch = ('x86', 'x64')[$Target64Bit]
+        UnixArch = ('x86', 'x86_64')[$Target64Bit]
+        CmakeArch = ('Win32', 'x64')[$Target64Bit]
+        Bits = ('32', '64')[$Target64Bit]
+        OutputPath = "${script:ProjectRoot}\windows\obs-${script:PackageName}-${script:Target}"
+    }
+
+    Log-Debug "
+Architecture    : $($script:ConfigData.Arch)
+CMake arch      : $($script:ConfigData.CmakeArch)
+Unix arch       : $($script:ConfigData.UnixArch)
+Target          : $($script:Target)
+Output dir      : $($script:ConfigData.OutputPath)
+Working dir     : $($script:WorkRoot)
+Project dir     : $($script:ProjectRoot)
+"
+}
+
+function Setup-BuildParameters {
+    if ( ! ( Test-Path function:Log-Output ) ) {
+        . $PSScriptRoot/Logger.ps1
+    }
+
+    $NumProcessors = (Get-CimInstance Win32_ComputerSystem).NumberOfLogicalProcessors
+
+    if ( $NumProcessors -gt 1 ) {
+        $env:UseMultiToolTask = $true
+        $env:EnforceProcessCountAcrossBuilds = $true
+    }
+
+    $VisualStudioData = Find-VisualStudio
+
+    $VisualStudioId = "Visual Studio {0} {1}" -f @(
+        ([System.Version] $VisualStudioData.Version).Major
+        ( $VisualStudioData.Name -split ' ')[3]
+    )
+
+    $script:CmakeOptions = @(
+        '-A', $script:ConfigData.CmakeArch
+        '-G', $VisualStudioId
+        "-DCMAKE_INSTALL_PREFIX=$($script:ConfigData.OutputPath)"
+        "-DCMAKE_PREFIX_PATH=$($script:ConfigData.OutputPath)"
+        "-DCMAKE_BUILD_TYPE=${script:Configuration}"
+        '--no-warn-unused-cli'
+    )
+
+    if ( $script:Quiet.isPresent ) {
+        $script:CmakeOptions += @(
+            '-Wno-deprecated', '-Wno-dev', '--log-level=ERROR'
+        )
+    }
+
+    Log-Debug @"
+
+Cmake options   : $($script:CmakeOptions)
+Multi-process   : ${NumProcessors}
+"@
+}
+
+function Find-VisualStudio {
+    <#
+        .SYNOPSIS
+            Find available Visual Studio instance
+        .DESCRIPTION
+            Uses WMI (Windows Management Instrumentation) to find an installed
+            Visual Studio instance on the host system
+        .EXAMPLE
+            Find-VisualStudio
+    #>
+
+    $VisualStudioData = Get-CimInstance MSFT_VSInstance
```

Comment:
Microsoft's Docs are unusually coy about that, I found suggestions that it can return _all_ available instances, but I haven't had time to create such an environment (especially as it isn't expected on CI). Given that the call returns a PowerShell object it should be easy to filter/adapt if that situation occurs.

Reply:
I guess we'll just leave this be until someone runs into it.